### ISSUE TITLE
legato + amp-velocity = clicks

### ISSUE THREAD

**Issue by TheWaveWarden** [TheWaveWarden/odin2#303]
When using Legato mode and the "Velocity" knob in the amplifier, note transition will produce clicks: This is due to the fact, that each note is scaled by it's velocity value, which abpruptly changes on note transition.

**Comment by xard-dev** [comment]
I can do a separate ticket for this but it would be almost duplicate to this issue is that legato + Rst = also clicks.

Snapping instrument can be made with very simple setup: 

1. set mode to legato
2. use one AnalogOsc in Sine mode
3. set octave on the AnalogOsc as low as it goes
4. add Highpass24 filter
5. set the filter around 300 Hz so that nothing can be heard when individual pressing keys
6. try pressing multiple keys in row holding first one down to get legato effect (still should not make much sound)
7. enable Rst from AnalogOsc
8. try pressing multiple keys in row again holding first one down

Now there should be a lot of snapping happening.
[Snappy Legato.zip](https://github.com/TheWaveWarden/odin2/files/5484080/Snappy.Legato.zip)

**Comment by TheWaveWarden** [comment]
@xard-dev Using Rst will reset the osc-phase to zero on each note-on event. Forcing a running sine wave to reset hard mid-cycle produces a click. This is kind of intended / expected (although I don't see a use-case for it really). Did you expect something else to happen in this case?

**Comment by xard-dev** [comment]
With Retrig and Poly modes this is quite expected. 

I can be wrong but I'd expect Legato to keep ADSR, LFO, oscillator positions and so on to just assign the ongoing note to new pitch when pushing more than one key at a time.

I know this is bit counter intuitive because this would mean that legato (when active) would ignore things like oscillator resets completely but that was my instinct.

**Comment by TheWaveWarden** [comment]
Yeah you're right about ADSR, LFO and so on. But I don't think Legato should ignore Rst, you always have the option to disable it. But then as said before Legato + Rst is an oddball combo anyways

**Comment by xard-dev** [comment]
You are right, it's quite counter intuitive and completely fine when creating new instrument while being aware of it. 

My only good point here would probably be when using complex preset it would be nice if the present wouldn't have to be modified specifically for legato to get it working in corner cases.
